### PR TITLE
Fixes VSTS Bug 681385: Accessibility: VoiceOver: Editor Caret does not

### PR DESCRIPTION
**Pull request by mkrueger**
follow VoiceOver

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/681385

### CODE REVIEW

File: main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs [hunk 1]
```diff
@@ -407,7 +407,7 @@ internal void Initialize (MonoTextEditor editor, TextDocument doc, ITextEditorOp
 			this.textEditorData.SelectionChanged += TextEditorDataSelectionChanged;
 			this.textEditorData.UpdateAdjustmentsRequested += TextEditorDatahandleUpdateAdjustmentsRequested;
 			Document.DocumentUpdated += DocumentUpdatedHandler;
-			
+			Document.EndUndo += delegate { HandleAccessibilityCursorChange (); };
```
Comment (slluis) [inline main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs]: Only called on EndUndo? shouldn't also be called when position or selection changes

File: main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs [hunk 2]
```diff
@@ -571,7 +571,99 @@ void CaretPositionChanged (object sender, DocumentLocationEventArgs args)
...
-		
+		ISegment lastAccessiblitySelection = null;
+		ITextSourceVersion lastAccessiblityVersion;
+		DocumentLocation oldAccessibilityCursorLocation;
+		static readonly string selectedTextMessage = GettextCatalog.GetString (", selected");
+		static readonly string unselectedTextMessage = GettextCatalog.GetString (", unselected");
+
+		void HandleAccessibilityCursorChange ()
+		{
+			void NewLineAnnouncement () { ShowAccessibilityAnnouncement (GettextCatalog.GetString ("New line")); }
```
Comment (slluis) [inline main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs]: This method should do nothing if accessibility or voiceover is disabled.

File: main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs]: Yes.

I need to test that.  It's called only on position / selection changes. However not on mouse selection yet but on keyboard commands.

File: main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs (hunk 2 above)
Comment (mkrueger) [reply main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs]: I need help from iain here atm it does no announcement - never.

**Comment by Therzok** [comment]
Not sure we still want a11y fixes for the old editor.

**Comment by mkrueger** [comment]
Closing - the old editor got replaced so there is no reason to work any longer on that.

**Pull request closed without merging**